django-rgd-imagery/rgd_imagery/models/processed.py

```diff
@@ -1,94 +1,53 @@
-        Image, on_delete=models.SET_NULL, null=True, related_name='+'
+    process_type = models.CharField(
+        max_length=20, default=ProcessTypes.ARBITRARY, choices=ProcessTypes.choices
     )
-    source_image = models.ForeignKey(Image, on_delete=models.CASCADE, related_name='+')
-
-    def _post_delete(self, *args, **kwargs):
-        # Cleanup the associated Image and ChecksumFile
-        if self.processed_image:
-            self.processed_image.file.delete()
-
-    permissions_paths = [('source_image', Image)]
-
+    parameters = models.JSONField(null=True, blank=True)
 
-class ConvertedImage(ProcessedImage):
-    """A model to store converted versions of an Image."""
+    # TODO: permissions_paths
 
-    task_funcs = (jobs.task_convert_to_cog,)
+    def _post_save(self, *args, **kwargs):
+        source_images = ProcessedImage.objects.filter(group=self)
+        for processed_image in source_images:
+            if processed_image.status not in [Status.QUEUED, Status.RUNNING]:
+                processed_image.save()
 
 
-class RegionImage(ProcessedImage):
-    """A sub region of an Image."""
+class ProcessedImage(TimeStampedModel, TaskEventMixin, PermissionPathMixin):
+    """Base class for processed images."""
 
-    task_funcs = (jobs.task_populate_region_image,)
+    task_funcs = (jobs.task_run_processed_image,)
 
-    class SampleTypes(models.TextChoices):
-        PIXEL_BOX = 'pixel box', _('Pixel bounding box')
-        GEO_BOX = 'geographic box', _('Geographic bounding box')
-        GEOJSON = 'geojson', _('GeoJSON feature')
-        ANNOTATION = 'annotation', _('Annotation entry')
+    source_images = models.ManyToManyField(Image, related_name='+')
 
-    sample_type = models.CharField(
-        max_length=20, default=SampleTypes.PIXEL_BOX, choices=SampleTypes.choices
+    processed_image = models.ForeignKey(
+        Image, on_delete=models.SET_NULL, null=True, blank=True, related_name='+'
     )
-    sample_parameters = models.JSONField()
-
-    def get_extent(self):
-        """Convert ``sample_parameters`` to length 4 tuple of XY extents.
-
-        Note
-        ----
-        A ``KeyError`` could be raised if the sample parameters are illformed.
+    ancillary_files = models.ManyToManyField(ChecksumFile, blank=True, related_name='+')
 
-        Return
-        ------
-        extents, projection: <left, right, bottom, top>, <projection>
-
-        """
-        p = self.sample_parameters
-
-        projection = p.pop('projection', None)
-        if self.sample_type in (
-            RegionImage.SampleTypes.PIXEL_BOX,
-            RegionImage.SampleTypes.ANNOTATION,
-        ):
-            projection = 'pixels'
+    def _pre_delete(self, *args, **kwargs):
+        if self.processed_image:
+            self.processed_image.file.delete()
+        # TODO: clean up ancillary_files - this throws an error when done through the admin interface
+        for file in self.ancillary_files.all():
+            file.delete()
```

Comment:
@mcovalt, can you help me figure out why this throws an error when deleting through the admin interface but not when calling `.delete()` on the object through the Python shell?

Reply:
I'm not getting an error, but I'm also not getting any files deleted. What was your error?

Reply:
What we really need is a FK w/ `CASCADE` from `ChecksumFile` to `ProcessedImage`. That would also enforce the constraint it seems we're assuming: one `ProcessedImage` per `ChecksumFile`.

Reply:
I wonder if Django will manage the `CASCADE` for us if we put the FK on a `ChecksumFile` we multi-table inherit it in our app and put the FK there?

Reply:
I'm very hesitant to start a multi-table inheritance for ChecksumFiles. We have relationships just like this one all over the place. Doing so would result in a lot of clutter, redundant tables, and just overall be hard to maintain. Let me double-check that this error is still occurring - I really think it shouldn't be and the current approach should suffice

Reply:
I'm just going to not implement this for now and leave it on the user to clean up ancillary files (which is what we're doing for `Raster`'s ancillary files field too.